### PR TITLE
Add basis, alias of fock

### PR DESCRIPTION
**Pull request by gautierronan**
Add `dq.basis` and `dq.basis_dm` to replicate the API of qutip. (I'm personally more used to using `qt.basis` so I keep making the mistake when using dynamiqs).

### CODE REVIEW

File: dynamiqs/utils/states.py [hunk 1]
```diff
@@ -97,6 +97,28 @@ def fock_dm(
...
 
 
+def basis(
+    dim: int | tuple[int, ...],
+    number: int | tuple[int, ...],
+    *,
+    dtype: torch.complex64 | torch.complex128 | None = None,
+    device: str | torch.device | None = None,
+):
+    """Alias for [`dq.fock()`](/python_api/utils/states/fock.html)."""
```
Comment (pierreguilmin) [inline dynamiqs/utils/states.py:107]: Correct syntax is
```suggestion
    """Alias for [`dq.fock()`][dynamiqs.fock]."""
```

**Review by pierreguilmin — CHANGES_REQUESTED** [review]
I'm good otherwise 👍

**Review by abocquet — APPROVED** [review]

File: dynamiqs/utils/states.py (hunk 1 above)
Comment (gautierronan) [reply dynamiqs/utils/states.py:107]: I tried this but the link was broken for some reason, both in the main function page or in the API summary table.

Comment (gautierronan) [reply dynamiqs/utils/states.py]: Nevermind it works if you use `[dynamiqs.fock]` and I was using `(dynamiqs.fock)`. Fixed.

**Review by abocquet — APPROVED** [review]

**Review by pierreguilmin — APPROVED** [review]

**Pull request merged**